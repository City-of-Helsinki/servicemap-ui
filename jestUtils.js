import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import english from './src/i18n/en';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: english,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });
