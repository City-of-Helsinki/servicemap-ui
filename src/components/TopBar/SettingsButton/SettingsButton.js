import React from 'react';
import PropTypes from 'prop-types';
import {
  Button, useMediaQuery,
} from '@mui/material';
import SettingsText from '../../SettingsText';

const SettingsButton = ({
  classes, settingsOpen, type, onClick,
}) => {
  const minWidth = useMediaQuery('(min-width:900px)');
  const maxWidth = useMediaQuery('(max-width: 1100px)');

  const isSelected = settingsOpen === type;

  const buttonClass = `
    ${classes.settingsButton} ${(isSelected && classes.settingsButtonPressed) || ''}
    ${(minWidth && maxWidth && classes.smallScreen) || ''}
  `;
  return (
    <Button
      aria-current={isSelected ? 'page' : false}
      aria-haspopup="dialog"
      id={`SettingsButton${type}`}
      classes={{ label: classes.buttonLabel }}
      className={buttonClass}
      onClick={onClick}
      role="button"
    >
      <SettingsText type={type} />
    </Button>
  );
};

SettingsButton.propTypes = {
  classes: PropTypes.shape({
    buttonLabel: PropTypes.string,
    settingsButton: PropTypes.string,
    settingsButtonPressed: PropTypes.string,
    smallScreen: PropTypes.string,
  }).isRequired,
  settingsOpen: PropTypes.oneOf(['citySettings', 'mapSettings', 'accessibilitySettings']),
  type: PropTypes.oneOf(['citySettings', 'mapSettings', 'accessibilitySettings']).isRequired,
  onClick: PropTypes.func.isRequired,
};

SettingsButton.defaultProps = {
  settingsOpen: null,
};

export default SettingsButton;
