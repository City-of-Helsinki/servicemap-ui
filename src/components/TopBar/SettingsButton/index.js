import { withStyles } from '@mui/styles';
import SettingsButton from './SettingsButton';
import styles from '../styles';

export default withStyles(styles)(SettingsButton);
