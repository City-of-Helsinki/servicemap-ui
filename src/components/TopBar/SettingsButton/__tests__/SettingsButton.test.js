import React from 'react';
import SettingsButton from '../SettingsButton';
import { getRenderWithProviders } from '../../../../../jestUtils';


const mockProps = {
  'aria-haspopup': 'dialog',
  settingsOpen: 'citySettings',
  type: 'citySettings',
  classes: {},
  onClick: () => {},
};

const renderWithProviders = getRenderWithProviders({
  settings: {
    visuallyImpaired: false,
    colorblind: false,
    hearingAid: false,
    mapType: 'servicemap',
    mobility: 'none',
    cities: 'helsinki',
  },
});

describe('<SettingsButton />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<SettingsButton {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does use correct aria arrtibutes', () => {
    const { container } = renderWithProviders(<SettingsButton {...mockProps} />);
    /*
      The following aria-attributes are based on the accessibility testing report from 26.4.2021
    */
    // Expect button role to be button
    expect(container.querySelector('button')).toHaveAttribute('role', 'button');
    // Expect button aria-haspopup value to be dialog
    expect(container.querySelector('button')).toHaveAttribute('aria-haspopup', 'dialog');
  });
});
