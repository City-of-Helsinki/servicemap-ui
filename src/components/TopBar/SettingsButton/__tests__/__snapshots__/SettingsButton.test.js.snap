// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsButton /> should work 1`] = `
<div>
  <button
    aria-current="page"
    aria-haspopup="dialog"
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root 
    undefined 
    
   css-1p2nkht-MuiButtonBase-root-MuiButton-root"
    id="SettingsButtoncitySettings"
    role="button"
    tabindex="0"
    type="button"
  >
    <p
      class="MuiTypography-root MuiTypography-subtitle1 SettingsText-title-1 css-dlbwt1-MuiTypography-root"
    >
      City
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 SettingsTextCurrentSettings SettingsText-text-3 css-1ak20dk-MuiTypography-root"
    >
      Show all
    </p>
  </button>
</div>
`;
